src/docs-developer-guide.rst

```diff
@@ -122,3 +122,188 @@ For example:
+  :param bool dotall: For using :py:data:`re.DOTALL`
+  :return: The error list consisting of error tuples: :py:const:`[(start, end, err, msg, replacement)]`.
+  :rtype: list
+
+.. py:function:: preferred_forms_check(text, list, err, msg, ignore_case=True, \
+                                       offset=0, max_errors=float("inf"))
+
+  To suggest a preferred form of the word used.
+  The parameter :py:data:`offset`
+  is not needed for style guide testing.
+
+
+  :param str text: Text to be checked
+  :param list list: list of comparison (words or regex): :py:const:`[correct form , incorrect form]`
+  :param str err: Name of the test
+  :param str msg: Error or warning message
+  :param bool ignore_case: For using :py:data:`re.IGNORECASE`
+  :param float max_errors: Maximum number of errors to be generated
+  :return: The error list consisting of error tuples: :py:const:`[(start, end, err, msg, replacement)]`.
+  :rtype: list
+
+
+.. py:function:: consistency_check(text, word_pairs, err, msg, offset=0)
+
+   To check for consistency for the given word pairs.
+   The parameters :py:data:`offset`
+   is not needed for style guide testing.
+
+   :param str text: Text to be checked
+   :param list word_pairs: Word pairs to be checked for consistency
+   :param str err: Name of the test
+   :param str msg: Error or warning message
+   :return: The error list consisting of error tuples: :py:const:`[(start, end, err, msg, replacement)]`.
+   :rtype: list
+
+.. note::
+
+  The checker functions are used by the
+  inbuilt proselint function :py:func:`lint`
+  to generate an error list of different format.
+  The returned list finally is: :py:const:`[(check, message, line, column, start, end, end - start, "warning", replacements)]`
+
+.. seealso::
+
+  To get a clear understanding,
+  refer the code for the modules `here <https://github.com/amperser/proselint/blob/master/proselint/tools.py>`_.
+
+.. rubric:: Example Usage
+
+.. code-block:: python
+
+  @memoize
+  def example(text):
+      """Example check."""
+      err = "style-guide.example"
+      msg = "A demonstration for writing checks."
+      regex = "[\.\?!](example)"
+
+      return existence_check(text, [regex], err, msg, ignore_case=False, 
+                         require_padding=False)
+
+When you define code-blocks which
+use inbuilt proselint testing,
+specify the class **style-checks**.
+
+.. code-block:: rst
+
+  .. code-block:: python
+    :class: style-checks
+
+The generated file after parsing code
+for style checks is :file:`style-checks.py`.
+
+
+If the test is too large
+to be defined in the file :file:`docs-style-guide.rst`,
+you can use a snippet from the test
+(as :ref:`here <american-spelling>`).
+The code-blocks for such snippets
+should specify the class **proselint-extra-checks**.
+Define the complete test
+in the file :file:`/style-guide/proselint-extra-checks.py`.
+
+.. _independent-checks:
+
+Independent checks
+~~~~~~~~~~~~~~~~~~~
+
```

Comment:
I'm unclear about proselint-independent checks. Do we have any? Can you provide more detail in this section?

Reply:
We have functions for checking quotes and section labels which have not used built-in functions of proselint to obtain an error list. Shall I add an example here?

Reply:
An example, and a explanation of what is meant, would be good.

Reply:
I have an added an example!